### PR TITLE
em100: init at unstable-2022-05-02-86631c9

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

The second commit adds a new package for the [`em100`](https://www.coreboot.org/EM100pro) tool written (mostly) by the coreboot team for the [Dediprog EM100Pro SPI Flash Emulator](https://www.dediprog.com/product/EM100Pro-G2).

The first commit adds myself (amjoseph) to `maintainers.nix`.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/drivers/em100/default.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  nativeBuildInputs = [
+    pkgconfig
+  ];
+
+  buildInputs = [
+    libusb
+    curl    # for optionally downloading firmware images
+  ];
+
+  installPhase = ''
```
Comment (Artturin) [inline pkgs/misc/drivers/em100/default.nix:28]: When overriding a phase remember to add the `runHooks for its pre and post phases`

File: pkgs/misc/drivers/em100/default.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{ lib
+, stdenv
+, curl
+, pkgconfig
+, libusb
+, fetchgit
+}:
+stdenv.mkDerivation rec {
+  pname = "em100";
+  version = "unstable-2022may02-86631c9";   # upstream does not appear to have formal releases
```
Comment (Artturin) [inline pkgs/misc/drivers/em100/default.nix:10]: We use ` = "unstable-2022-05-02";` in nixpkgs

File: pkgs/misc/drivers/em100/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
+{ lib
+, stdenv
+, curl
+, pkgconfig
```
Comment (Artturin) [inline pkgs/misc/drivers/em100/default.nix:4]:
```suggestion
, pkg-config
```

pkgconfig is a alias to pkg-config

File: pkgs/misc/drivers/em100/default.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      - Adjust the state of the hold pin, which supports overriding the internal SPI
+      - Use of several em100 devices, distinguished by their serial number
+      - Terminal mode, allowing the SoC to send messages
+      - Output a trace of SPI commands issued by the SoC
+      - Reading / writing em100 firmware (dangerous as it can brick your em100)
+
+      [1] https://www.dediprog.com/product/EM100Pro-G2
+      '';
+    license = licenses.gpl2Plus;
+    sourceProvenance = [ ];
```
Comment (Artturin) [inline pkgs/misc/drivers/em100/default.nix:62]: Fill?

Comment (amjoseph-nixpkgs) [reply pkgs/misc/drivers/em100/default.nix]: `sourceProvenance=[]` and `sourceProvenance=[lib.sourceTypes.fromSource]` [mean the same thing](https://github.com/NixOS/nixpkgs/blob/ef3fe254f3c59455386bc92fe84164f9679b92b1/doc/stdenv/meta.chapter.md?plain=1#L257).  Yeah, I think that's weird.

**Comment by amjoseph-nixpkgs** [comment]
Force-push is a squash, no other changes.

**Comment by amjoseph-nixpkgs** [comment]
Second force-push fixes the version number in the commit message, no other changes.

**Comment by Artturin** [comment]
I meant to merge this once the ci was done but I forgot 😅

**Pull request closed without merging**